src/services/codefixes/importFixes.ts

```diff
@@ -329,7 +329,9 @@ namespace ts.codefix {
 
             const relativePath = removeExtensionAndIndexPostFix(getRelativePath(moduleFileName, sourceDirectory, getCanonicalFileName), options);
             if (!baseUrl) {
-                return [relativePath];
+                // Don't use a relative path that would pass through node_modules -- should have handled that that in `tryGetModuleNameAsNodeModule`,
+                // but would do nothing there if we're in classic resolution, so return no result in that case.
+                return forEachAncestorDirectory(relativePath, p => getBaseFileName(p) === "node_modules" ? true : undefined) ? [] : [relativePath];

```

Comment:
why not just check module resolution strategy is classic resolver when doing this?

Reply:
Actually, now I'm thinking we shouldn't do anything here.
We're providing an import fix for that file because the file has already been included in the project. In the case of this test, that's because fourslash adds all the files to the language service. In a real scenario that would only happen if the file had already been imported somehow.
So I think it would suffice to just test that the import fix is for a relative path and not a global import. Does that sound right?